src/System.Windows.Forms.Primitives/src/Interop/Ole32/Interop.GPStream.cs

```diff
@@ -85,33 +104,36 @@ public unsafe void CopyTo(IStream pstm, ulong cb, ulong* pcbRead, ulong* pcbWrit
 
                 if (pcbWritten is not null)
                     *pcbWritten = totalWritten;
+
+                return HRESULT.S_OK;
             }
 
-            public unsafe void Read(byte* pv, uint cb, uint* pcbRead)
+            HRESULT Com.ISequentialStream.Interface.Read(void* pv, uint cb, uint* pcbRead)
             {
+                if (pv is null)
+                {
+                    return HRESULT.STG_E_INVALIDPOINTER;
+                }
+
                 ActualizeVirtualPosition();
 
-                Span<byte> buffer = new Span<byte>(pv, checked((int)cb));
+                Span<byte> buffer = new(pv, checked((int)cb));
                 int read = _dataStream.Read(buffer);
 
                 if (pcbRead is not null)
                     *pcbRead = (uint)read;
-            }
 
-            public void Revert()
-            {
-                // We never report ourselves as Transacted, so we can just ignore this.
+                return HRESULT.S_OK;
             }
 
-            public unsafe void Seek(long dlibMove, SeekOrigin dwOrigin, ulong* plibNewPosition)
-            {
-                long position = _virtualPosition;
-                if (_virtualPosition == -1)
-                {
-                    position = _dataStream.Position;
-                }
+            HRESULT Com.IStream.Interface.Read(void* pv, uint cb, uint* pcbRead) =>
+                ((Com.ISequentialStream.Interface)this).Read(pv, cb, pcbRead);
```

Comment:
It's customary to have `=>` on the new line
```suggestion
            HRESULT Com.IStream.Interface.Read(void* pv, uint cb, uint* pcbRead)
                => ((Com.ISequentialStream.Interface)this).Read(pv, cb, pcbRead);
```